openmmmcmc/states.py

```diff
@@ -0,0 +1,1903 @@
+                try:
+                    context.setParameter(barostat.Temperature(), self.temperature)
+                except AttributeError:  # OpenMM < 7.1
+                    openmm_state = context.getState(getPositions=True, getVelocities=True,
+                                                    getParameters=True)
+                    context.reinitialize()
+                    context.setState(openmm_state)
+        elif self._barostat is not None:
+            # The context is NVT but this is NPT.
+            raise ThermodynamicsError(ThermodynamicsError.INCOMPATIBLE_ENSEMBLE)
+
+        # Apply temperature to thermostat or integrator.
+        thermostat = self._find_thermostat(system)
+        if thermostat is not None:
+            if thermostat.getDefaultTemperature() != self.temperature:
+                thermostat.setDefaultTemperature(self.temperature)
+                context.setParameter(thermostat.Temperature(), self.temperature)
+        else:
+            integrator = context.getIntegrator()
+            self._set_integrator_temperature(integrator)
+
+    # -------------------------------------------------------------------------
+    # Internal-usage: system handling
+    # -------------------------------------------------------------------------
+
+    # Standard values are not standard in a physical sense, they are
+    # just consistent between ThermodynamicStates to make comparison
+    # of standard system hashes possible. We set this to round floats
+    # and use OpenMM units to avoid funniness due to precision errors
+    # caused by periodic binary representation/unit conversion.
+    _STANDARD_PRESSURE = 1.0*unit.bar
+    _STANDARD_TEMPERATURE = 273.0*unit.kelvin
+
+    _NONPERIODIC_NONBONDED_METHODS = {openmm.NonbondedForce.NoCutoff,
+                                      openmm.NonbondedForce.CutoffNonPeriodic}
+
+    def _check_internal_consistency(self):
+        """Shortcut self._check_system_consistency(self._system)."""
+        self._check_system_consistency(self._system)
+
+    def _check_system_consistency(self, system):
+        """Check system consistency with this ThermodynamicState.
+
+        Raise an error if the system is inconsistent. Currently checks
+        that there's 1 and only 1 thermostat at the correct temperature,
+        that there's only 1 barostat (or none in case this is in NVT),
+        that the barostat is supported, has the correct temperature and
+        pressure, and that it is not associated to a non-periodic system.
+
+        Parameters
+        ----------
+        system : simtk.openmm.System
+            The system to test.
+
+        Raises
+        ------
+        ThermodynamicsError
+            If the system is inconsistent with this state.
+
+        """
+        TE = ThermodynamicsError  # shortcut
+
+        # This raises MULTIPLE_THERMOSTATS
+        thermostat = self._find_thermostat(system)
```

Comment:
This function only check the `System`. The integrator is in the `Context` object, so this problem is handled only in `create_context(integrator, platform)` and `apply_to_context(context)`